lib/oli/delivery/attempts/activity_lifecycle/evaluate.ex

```diff
@@ -501,7 +493,7 @@ defmodule Oli.Delivery.Attempts.ActivityLifecycle.Evaluate do
     case Oli.Activities.get_registration_by_slug(activity_registration_slug) do
       %Oli.Activities.ActivityRegistration{allow_client_evaluation: true} ->
         Repo.transaction(fn ->
-          part_attempts = get_latest_part_attempts(activity_attempt_guid)
+          part_attempts = part_attempts_input || get_latest_part_attempts(activity_attempt_guid)
```

Comment:
Here, the value of `part_attempts_input` received in the function will be used and, in case it's nil, the function `get_latest_part_attempts/1` will be called, I'm right?

Reply:
Exactly! 

The `||` operator accept any type (as opposite of `or`), and `nil` is falsy so evaluates to false, so as you mention, in that case will return what `get_latest_part_attempts/1` has